lib/puppet/parser/functions/defined_with_params.rb

```diff
@@ -24,7 +24,19 @@
     params = {}
   end
   ret = false
-  if resource = findresource(reference.to_s)
+
+  # Workaround for PE-20308
+  if reference.is_a?(String)
+    type_name, title = Puppet::Resource.type_and_title(reference, nil)
```

Comment:
I'll just fallback to the old behavior pre 4.5.0 as the ticket only reports the behavior on newer puppet versions.